cmd/buildctl/build.go

```diff
@@ -291,7 +321,20 @@ func buildAction(clicontext *cli.Context) error {
 		return pw.Err()
 	})
 
-	return eg.Wait()
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	if txt, ok := subMetadata["result.txt"]; ok {
```

Comment:
Would we want a case for `result.json` as well? Feels like we could format it here, in case the frontend hasn't done it.

Reply:
We would need a specific printer for this and I don't want `buildctl` to have too much hardcoded info about specific requests.

Or are you saying that if there is a `result.json` we should only print that directly and not the full result like it is in current commit.

Reply:
I guess we have an exception for `result.txt`, I'm just wondering why we shouldn't also have one for `result.json`. I think at the least, we should print directly and not the whole result if we find a `result.json` similar to `result.txt`.

Although maybe for buildctl, since it's a dev tool, we should assume no knowledge of the response, and always print out everything?

Reply:
> and always print out everything?

That what it does atm. There is a special case for `result.txt`, so if it exists then the expectation is that frontend has already formatted the result and we can just print that output. Otherwise we just print whatever frontend provided(including result.json). If it makes more sense I can make it so that it only prints `result.json` if it exists and skips the rest.